### PR TITLE
Fix Alamut link for MT variants in build 38

### LINKED ISSUE #4150: Broken alamut link to mitochondrial genes on GRHc38
The alamut link in scout is not working properly for variants in mitochondrial genes. This is caused by the gene prefix was changed from **M** to **MT** with GRHc38.

Current alamut url 
`http://localhost:10000/search?apikey=73435198&institution=ASE0046&request=M(GRCh38):3116C%3ET`

Working URL,
`http://localhost:10000/search?apikey=73435198&institution=ASE0046&request=MT(GRCh38):3116C%3ET`

**Suggested solution**

Implement a lookup table with synonymous names so **M** and **MT** are valid prefixes.

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds a functionality or fixes a bug.
- Fix #4150

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. Deploy this branch and test connecting to Alamut using a MT variant in a case analysed with genome build 38

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [x] code approved by Jakob37
- [x] tests executed by Jakob37

### CODE REVIEW

**Comment by northwestwitch** [comment]
@mhkc @alkc sorry for the delay with this fix. This is a bit hard to test on our server (we don't have any cases on build 38), but if you have time you could test it and review the code? Thanks!

File: scout/server/links.py [hunk 1]
```diff
@@ -751,29 +751,32 @@ def alamut_link(
...
     search_verb = "search" if alamut_key else "show"
     alamut_key_arg = f"apikey={alamut_key}&" if alamut_key else ""
     alamut_inst_arg = f"institution={alamut_institution}&" if alamut_institution else ""
+    chromosome = variant_obj.get("chromosome")
+
+    build = build or 37
+    build_str = ""
+    if build == 38:
+        build_str = "(GRCh38)"
+        chromosome.replace("M", "MT")
```
Comment (Jakob37) [inline scout/server/links.py:773]: Should it be `chromosome = chromosome.replace("M", "MT")` ?

Testing it in the terminal:

```
>>> chromosome = "M"
>>> chromosome.replace("M", "MT")
'MT'
>>> chromosome
'M'
```

Comment (northwestwitch) [reply scout/server/links.py:773]: Rigth, thanks!

**Review by Jakob37 — COMMENTED** [review]
With `chromosome = ` updated, it worked for the Lund-geneticists!

**Review by Jakob37 — APPROVED** [review]

File: scout/server/links.py (hunk 1 above)
Comment (northwestwitch) [reply scout/server/links.py]: Fixed! please try again now

Comment (Jakob37) [reply scout/server/links.py]: The `git diff` is identical to the fix-branch I tested, so this should be fine!

Comment (northwestwitch) [reply scout/server/links.py]: Thank you for testing and the review. The fix will be included in the next Scout release!

**Pull request merged**